functions.md

```diff
@@ -0,0 +1,131 @@
+- The name of the function, or its identifier, followed by parentheses.
+- A function body, or the block of statements required to perform a specific task, enclosed in the function’s curly brackets, { }.
+
+### Calling a function
+A function is called when `function` name followed by the parentheses is typed. From the example above
+
+```js
+console.log(teamNames());
+```
+called the function inside `teamNames`.
+
+### Parameter and Argument
+parameter allows function accept input(s), and perform a task using the input(s). When declaring a function we can specify a parameter.  We use parameters as placeholders for information that will be passed to the function when it is called. example:
+
+```js
+function aggregateScore(Home, Away) {
+    console.log(Home + Away);
+}
+```
+the value passed to the functiion when they are called, are called `argument`. it can be passed as a value or variable example:
+
+```js
+function aggregateScore(Home, Away) {
+    console.log(Home + Away);
+}
+aggregateScore(4 + 2)// 6
+```
+
+### Default parameters
+in `ES6` default parameter feature is included, with the ability to have a predetermined value, in case  there is no argument passed into function or the argument is `undefined` when called. example:
+
+```js
+function greeting (name = 'stranger') {
+  console.log(`Hello, ${name}!`)
+}
+
+greeting('Nick') // Output: Hello, Nick!
+greeting() // Output: Hello, stranger!
+```
+By using a default parameter, we account for situations when an argument isn’t passed into a function that is expecting an argument.
+
+### Return
+when a function is called, the computer will run through the functions code and evaluate the result of calling the function. By default that resulting value is undefined. example:
+
+```js
+function monitorCount(rows, columns) {
+   let total = rows + columns
+}
+console.log(monitorCount(3 + 5));// print undefined
+```
+
+ the `return` keyword allow us see the calculation that the computer did print example:
+
+```js 
+function totalNumber(cash, bank) {
+    return 30 +80;
+}
+console.log(totalNumber()); // 11
+```
+
+The `return` keyword is powerful because it allows functions to produce an output. We can then save the output to a variable for later use
+
+### Function Expression
+ Another way of defining a function is to use a function expression. example:
+ 
+ ```js
+ const myName = Function(boy {
+
+ })
+  ```
+  To declare a function expression:
+
+- Declare a variable to make the variable’s name be the name, or identifier, of your function. Since the release of ES6, it is common practice to use const as the keyword to declare the variable.
+
+- Assign as that variable’s value an anonymous function created by using the function keyword followed by a set of parentheses with possible parameters. Then a set of curly braces that contain the function body. Unlike function declarations, function expressions are not hoisted so they cannot be called before they are defined.
+
+  ### Arrow Function
```

Comment:
Check this heading, seems not to be properly formatted.

Reply:
i think it's okay!!!

Reply:
Can you explain why you using it as a quoted heading? The spaces before the heading tag makes it look like a quoted heading, is that your intented result?

Reply:
It's a mistake, it's now adjusted. Thanks